### ISSUE TITLE
As a User I want Game On! to log me in so I can authenticate future API requests

### ISSUE THREAD

**Issue by olafwrieden** [DanielLarsenNZ/gameon#38]
This is the frontend story for #4.

**Comment by olafwrieden** [comment]
@DanielLarsenNZ I have both client and server running and am testing your included `wwwroot/index.html` (via Live Server). The generic modal pops up.. do I need a demo account that you have registered in our tenant's AAD? I've lost access to my alias last week so haven't received any auth details in case you PM'd them :(

**Comment by DanielLarsenNZ** [comment]
Oh no! When do you get your new alias? This may be a good chance to get multi-tenancy working; do you have an AAD tenant of your own? If not you can use mine.

**Comment by olafwrieden** [comment]
Most likely not until late January. I've experimented with my own tenant by following [this guide](https://medium.com/@pavelray/connect-your-react-app-with-azure-ad-using-3ddd39223d27), yet was unable to sign in despite creating a new user and registering the app. I wonder if you could create me an account in your tenant and [email me](mailto:olaf@olafwrieden.com) the details.

I've done some light reading up on multi-tenancy and think it will set us up for future features. With that, I've created one frontend (#39) and one backend (#40) story to cover owners creating and players joining 'public' competitions respectively.

**Comment by DanielLarsenNZ** [comment]
Can you email me please? daniel@larsen.nz

**Comment by DanielLarsenNZ** [comment]
OK, I have done a bunch of work here and now have a but more of an understanding of how AAD Apps should work in this scenario. 

I need you to make a couple of changes:

* The clientId for your frontend app is now `api://GameOn.Api`
* And for now the requested scopes should be just one: "api://GameOn.Api/Users"

Here is a full example: [index.html](https://github.com/DanielLarsenNZ/gameon/blob/main/server/GameOn/GameOn.Auth/wwwroot/index.html#L14-L28)

I don't think this will work with msal.js v1, so let me know if you need help to get msal.js v2 working. 

The Users service now has the ability to call the Graph on behalf of the user to get User photos. I will comment about this in the related issues. From implementing this I have made several changes to the User model, including removing the Email and ImageUrl properties which are no longer required. 

Apologies this has taken so long. This has been a very frustrating journey as the AAD B2B story has changed so much recently.

**Comment by olafwrieden** [comment]
Fantastic! I got MSAL (v2) working last week. It's so easy with `@azure/msal-browser` and `@azure/msal-react` (though still in Alpha as pointed out earlier).

I successfully logged in from another tenant and it gave me the bearer token as expected... didn't commit yet because I found some edge cases where if the user undergoes an SSO auth experience, they were not always redirected to the correct page on the frontend - which would cause user data (ie. a call to `GET /me`) to become out of sync with what was previously in the application state. Small problem with application-wide consequences but something I want to get working correctly asap.

I don't have a lot of time this week but will do what I can. Thank you for capturing your findings in the docs! It really helped me fill in some blanks and understand MSAL better! Keep it up!
